packages/states-plugin/src/main/statesPlugin.ts

```diff
@@ -0,0 +1,67 @@
+import { Field, Plugin } from 'roqueform';
+
+export interface StatesPlugin {
+  /**
+   * Returns `true` if the field value or derived fields are different from the value it was initialized with. `false` otherwise.
+   *
+   * **Note:**
+   * Comparison is done with reference-equals.
+   * For the detecting dirty, fields need to be created and initialized before first dispatch value.
+   */
+  isDirty(): boolean;
+
+  /**
+   * Recursively deletes dirty flags associated with this field and all of its derived fields.
+   * Also resets values to initial ones.
+   */
+  reset(): void;
+}
+
+/**
+ * By default, comparison is done with reference-equals.
+ *
+ * Used for detecting is field dirty.
+ */
+export type EqualityChecker = (left: any, right: any) => any;
```

Comment:
A type cannot know what default value it has on the usage side.

```suggestion
/**
 * Returns truthy result if values are equal, and falsy result otherwise.
 */
export type EqualityChecker = (left: any, right: any) => any;
```